Source/CarthageKit/GitURL.swift

```diff
@@ -47,21 +47,27 @@ public struct GitURL {
 
 	/// The name of the repository, if it can be inferred from the URL.
 	public var name: String? {
-		let absoluteURLString: String
-		if urlString.hasPrefix(".") {
-			absoluteURLString = URL(fileURLWithPath: FileManager.default.currentDirectoryPath)
-				.appendingPathComponent(urlString)
-				.standardizedFileURL
-				.absoluteString
-		} else {
-			absoluteURLString = urlString
-		}
-		let components = absoluteURLString.split(omittingEmptySubsequences: true) { $0 == "/" }
-
-		return components
+		let lastComponent = urlString
+			.replacingOccurrences(of: "\u{0000}", with: "\u{2400}") // can’t have those
+			.split(omittingEmptySubsequences: true) { $0 == "/" }
 			.last
 			.map(String.init)
-			.map(strippingGitSuffix)
+
+		precondition(lastComponent != .some("")) /* because of `omittingEmptySubsequences` */
+		
+		/// Potentially used to prevent backwards or noop directory traversal via «FULL STOP» characters…
+		/// …by deploying the «FULLWIDTH FULL STOP» character.
+		var replacementForEntirelyCharactersOfFullStop: [Character] = []
+		for char in (lastComponent ?? "") {
+			guard char == "." else { replacementForEntirelyCharactersOfFullStop = []; break }
+			replacementForEntirelyCharactersOfFullStop.append("\u{FF0E}")
```

Comment:
imho this is confusing as result, maybe just replace with `_` ?

Reply:
clarified over slack

Reply:
So, that character, as an alternative to the «FULLWIDTH FULL STOP» character:

The «FULLWIDTH FULL STOP» character has a width to it (in macOS Terminal’s default state, in Finder, in others) that makes it noticeably different.

IMO that ‘different, but able to correspond it to what’s listed in a Cartfile’ nature give it utility. Since you mentioned you're good with the PR in overall approval, let’s get it merged 🚀